### DISCUSSION TITLE
Using url to analog example leaves speaker "bugged" repeating last fragment

### DISCUSSION THREAD

**Discussion by FGAC1226** [pschatzmann/arduino-audio-tools#693] [Q&A] [answered]
I am using the "Streaming Radio Player" example (https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-stream/streams-url_mp3-analog) to play a text-to-speech mp3 file from a server, and it's all working almost as expected, except that after the mp3 is finished, the speaker keeps playing the last half-second or so over and over on repeat.
The expected behavior is for the speaker to just go silent.

Additionally, in the example, "copier.copy();" is just left on the "void loop". Would a 
```
//Some Code
while(copier.copy()){
copier.copy();
}
//More Code
```
make the code continue after the file stops playing? Or is there another recommended way? 
Thank you very much

The code being used:

```
/**
 * @file streams-url_mp3-analog.ino
 * @author Phil Schatzmann
 * @brief decode MP3 stream from url and output it on I2S
 * @version 0.1
 * @date 2021-96-25
 * 
 * @copyright Copyright (c) 2021
 */

// install https://github.com/pschatzmann/arduino-libhelix.git

#include "AudioTools.h"
#include "AudioCodecs/CodecMP3Helix.h"


URLStream url("Airbox-9D6D","75809058");
AnalogAudioStream analog; // final output of decoded stream
EncodedAudioStream dec(&analog, new MP3DecoderHelix()); // Decoding stream
StreamCopy copier(dec, url); // copy url to decoder


void setup(){
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);  

  // setup analog
  auto config = analog.defaultConfig(TX_MODE);
  analog.begin(config);

  // setup I2S based on sampling rate provided by decoder
  dec.setNotifyAudioChange(analog);
  dec.begin();

// mp3 radio
  url.begin("http://georgiou.pythonanywhere.com/wav","audio/mp3");
 
}

void loop(){
  copier.copy();
}
```

**Comment by pschatzmann** [comment, marked as answer]
I am not sure where this would come from. In any case I could never reproduce this myself, so it could be related to your hardware.
There are plenty of options that you can try:
1) Use a wav file instead of a mp3
2) instead of copy in the loop just do a copyAll() in the setup
3) when the copy is done (if copier.copy()==0) you could 
3.1) close the output: analog.end()
3.2) just continue to write silence: output.writeSilence(1024)

**Reply by FGAC1226** [reply]
Many thanks for your answer, I tried keeping it as a mp3 file and using analog.end(), which works "well" (now instead of repeating it forever, it's just a couple of times).
I would use it as a wav, but for some reason, when it's a wav file no audio plays.
I think this might be related to the server-side though, since when manually visiting the web it seems that if the file plays natively in the browser, the arduino plays it without issue, but when it instead auto-downloads, no sound comes from the arduino.

In the code I just changed "audio/mp3" to "audio/wav".

Thabk you for your help, and for providing and maintaining such an amazing library.

**Comment by pschatzmann** [comment]
I added the auto_clear attribute to the config. It might help to set config.auto_clear = true;
I was running into the same issue in the I2SStream when this was not set to true...
